### DISCUSSION TITLE
(ANSWERED) Epoch in position_of_radec

### DISCUSSION THREAD

**Discussion by Flapnuggets** [skyfielders/python-skyfield#627] [Q&A]
Is this the proper way to handle precession?

```
t = ts.utc(year, month,day,0)
astrometric = earth.at(t).observe(orbit)
ra, dec, distance = astrometric.radec()
hours, mins, secs = str(ra).split(' ')
hours = float(hours.replace('h', ''));
degs, dmins, dsecs = str(dec).split(' ')
degs = int(degs.replace('deg', ''))
dist, au = str(distance).split(' ')
dist = float(dist)
orbloc = position_of_radec(hours, degs, dist, epoch=t)
constellation = const_at(orbloc)
```

How could I add in a geoLocation to the observation point as well?

**Comment by brandon-rhodes** [comment]
For future readers who might happen across this question, I think it wound up being fully answered in an accompanying issue #628 but here’s a quick summary for folks who might be sent here by a search:

* Happily, you should never need to parse numbers out of strings in Skyfield! If you find yourself doing so, look again in the documentation. If that doesn't work, ask, and we'll try to make the documentation clearer.
* In this case `ra.hours` and `distance.au` return the two floating point numbers that the above code tries to parse out of strings — losing many digits of precision in the process.
* But the above code doesn’t actually need `ra.hours` and `distance.au` because the call to `position_of_radec()` isn’t necessary. If you want RA and dec expressed using a distant equinox instead of J2000, simply add that date to your call: `.radec(t)`.
